### ISSUE TITLE
Multi GPU not working anymore?

### ISSUE THREAD

**Issue by oliverban** [oobabooga/text-generation-webui#2715]
### Describe the bug

Hello!

I haven't used text generation webui in about 3 weeks. Updated yesterday and started my default 30B-GPTQ model. I did however notice right away that the model is loaded only on a single device, when before the update it would split the model on both my 3090's.

When checking "Auto-Devices" in either "Model" or writing it in the start cmd like "--auto-devices" WEBUI fails to start/load the model. It just sits there. I've tried setting all the Memory sliders (gpu0, gpu1 and cpu) to the same amount and also 0 but nothing works. Can't load it on both.

I did search for issues but couldn't find anything recent.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Start webui on PC with two (2) GPUs and watch as WEBUI ignores the second one when loading the model.

### Logs

```shell
bin C:\Users\Oliver\Documents\Github\oobabooga-windows\installer_files\env\lib\site-packages\bitsandbytes\libbitsandbytes_cuda117.dll
The following models are available:

1. decapoda-research_llama-7b-hf
2. mosaicml_mpt-7b-storywriter
3. notstoic_pygmalion-13b-4bit-128g
4. Pygmalion6b
5. TheBloke_guanaco-33B-GPTQ
6. TheBloke_wizard-mega-13B-GPTQ-4bit-128g
7. TheBloke_Wizard-Vicuna-13B-Uncensored-GPTQ-4bit-128g
8. TheBloke_Wizard-Vicuna-30B-Uncensored-GPTQ-4bit
9. WizardLM-7B-Uncensored

Which one do you want to load? 1-9

8

2023-06-16 13:47:49 INFO:Loading TheBloke_Wizard-Vicuna-30B-Uncensored-GPTQ-4bit...
2023-06-16 13:47:49 INFO:The AutoGPTQ params are: {'model_basename': 'Wizard-Vicuna-30B-Uncensored-GPTQ-4bit--1g.act.order', 'device': 'cuda:0', 'use_triton': False, 'use_safetensors': True, 'trust_remote_code': False, 'max_memory': None, 'quantize_config': None}
2023-06-16 13:47:53 WARNING:The safetensors archive passed at models\TheBloke_Wizard-Vicuna-30B-Uncensored-GPTQ-4bit\Wizard-Vicuna-30B-Uncensored-GPTQ-4bit--1g.act.order.safetensors does not contain metadata. Make sure to save your model with the `save_pretrained` method. Defaulting to 'pt' metadata.
2023-06-16 13:48:07 WARNING:skip module injection for FusedLlamaMLPForQuantizedModel not support integrate without triton yet.
2023-06-16 13:48:07 INFO:Loaded the model in 18.08 seconds.

2023-06-16 13:48:07 INFO:Loading the extension "gallery"...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
```

### System Info

```shell
SysRAM: 64gb
VRAM: 24 (48 in total)
GPUs: 2 x 3090s
CPU: AMD Threadripper 16 cores, 32 threads.
```

**Comment by Ph0rk0z** [comment]
Don't set auto devices. The default loader is now autogptq. Just set the memory split from the command line (or UI).

Consider exllama and llama_inference_offload as well vs autogptq. Much better perf. As you've been running it, I think you might be hurting speed by using 2 cards on a only a 30b 4 bit with no group size.

**Comment by oliverban** [comment]
> Don't set auto devices. The default loader is now autogptq. Just set the memory split from the command line (or UI).
> 
> Consider exllama and llama_inference_offload as well vs autogptq. Much better perf. As you've been running it, I think you might be hurting speed by using 2 cards on a only a 30b 4 bit with no group size.

Sounds great. Even when not having --auto-devices it doesn't work on two cards. Should I set the group size and bits manually in the web ui and save settings with only memory split made? Does that make the 2 cards work in tandem again? Any help here would be appreciated, don't want to lose out on performance!

**Comment by Ph0rk0z** [comment]
The GPU split of exllama definitely works because I use it for the 65b. It probably fills the first card and has nothing to put on the second. 

The memory limits have only ever been a suggestion to accelerate and transformers. You aren't losing anything by it working on one card. You're gaining. Plus all the context can only go on one anyway. Hence you split 16g to one GPU to leave room for it.

Convert the 33b to 8bit if you want to use 2 cards. It can be GPTQ or BnB. Or just d/l the 65b.

**Comment by oliverban** [comment]
> 

Yeah, seems like it everything is working now, but the split when using exllama. But might be as you say, that it doesn't need the second card.

But yeah, speeds are WAY different haha! Exllama is blazing!
